src/VisualStudio/CSharp/Impl/Debugging/LocationInfoGetter.cs

```diff
@@ -1,94 +1,155 @@
+            var memberDeclaration = token.GetAncestor<MemberDeclarationSyntax>();
+
+            if (memberDeclaration == null)
+            {
+                return default(DebugLocationInfo);
+            }
 
             // field or event field declarations may contain multiple variable declarators. Try finding the correct one.
             // If the position does not point to one, try using the first one.
-            if (memberDecl != null &&
-                (memberDecl.Kind() == SyntaxKind.FieldDeclaration || memberDecl.Kind() == SyntaxKind.EventFieldDeclaration))
+            VariableDeclaratorSyntax fieldDeclarator = null;
+            if (memberDeclaration.Kind() == SyntaxKind.FieldDeclaration || memberDeclaration.Kind() == SyntaxKind.EventFieldDeclaration)
             {
-                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDecl).Declaration.Variables;
+                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDeclaration).Declaration.Variables;
 
                 foreach (var declarator in variableDeclarators)
                 {
                     if (declarator.FullSpan.Contains(token.FullSpan))
                     {
-                        memberDecl = declarator;
+                        fieldDeclarator = declarator;
                         break;
                     }
                 }
 
-                if (memberDecl == null)
+                if (fieldDeclarator == null)
                 {
-                    memberDecl = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
+                    fieldDeclarator = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
                 }
             }
 
-            if (memberDecl != null)
-            {
-                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl, cancellationToken);
+            var name = GetName(memberDeclaration, fieldDeclarator);
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var lineNumber = text.Lines.GetLineFromPosition(position).LineNumber;
+            var accessor = token.GetAncestor<AccessorDeclarationSyntax>();
+            var memberLine = text.Lines.GetLineFromPosition(accessor?.SpanStart ?? memberDeclaration.SpanStart).LineNumber;
+            var lineOffset = lineNumber - memberLine;
+
+            return new DebugLocationInfo(name, lineOffset);
+        }
 
-                if (memberSymbol != null)
+        private static string GetName(MemberDeclarationSyntax memberDeclaration, VariableDeclaratorSyntax fieldDeclaratorOpt)
+        {
+            const string missingInformationPlaceholder = "?";
+
+            ArrayBuilder<string> containingDeclarationNames = null;
```

Comment:
In VB, this is also hit if you have a breakpoint on a field initializer in a class with no containing namespace.